include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_histogram.h

```diff
@@ -0,0 +1,564 @@
+// -*- C++ -*-
+//===-- parallel_backend_sycl_histogram.h ---------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_PARALLEL_BACKEND_SYCL_HISTOGRAM_H
+#define _ONEDPL_PARALLEL_BACKEND_SYCL_HISTOGRAM_H
+
+#include <algorithm>
+#include <cstdint>
+#include <type_traits>
+
+#include "sycl_defs.h"
+#include "parallel_backend_sycl_utils.h"
+#include "execution_sycl_defs.h"
+#include "unseq_backend_sycl.h"
+#include "utils_ranges_sycl.h"
+
+#include "../../histogram_binhash_utils.h"
+#include "../../../internal/async_impl/async_impl_hetero.h"
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace __par_backend_hetero
+{
+
+template <typename _Range>
+struct __custom_boundary_range_binhash
+{
+    _Range __boundaries;
+    __custom_boundary_range_binhash(_Range __boundaries_) : __boundaries(__boundaries_) {}
+
+    template <typename _T2>
+    ::std::int32_t
```

Comment:
=> auto ?
(or some _BinIndex)

Reply:
I've updated all these `get_bin` returns to be `auto` where it makes sense.